analyses/cell-type-ewings/cell_type_annotations/SCPCS000492/SCPCL000824_tumor-gene-expression-exploration.Rmd

```diff
@@ -0,0 +1,670 @@
+                      names_to = "gene_list", 
+                      values_to = "signature_score") |> 
+  dplyr::mutate(
+    gene_list = stringr::word(gene_list, 1, sep = "_")
+  )
+```
+
+```{r}
+ggplot(ucell_df, aes(x = signature_score, colour = sample)) +
+  geom_density() +
+  facet_grid(rows = vars(gene_list))
+```
+
+Here we see that the marker gene set maybe has a small bimodal distribution in `SCPCL000822`, but it is very small and not quite as striking as looking at the scores from `AUCell` or raw gene expression. 
+`RIGGI`, `ZHANG`, and `SILIGAN` both show very similar distributions to the gene set scores distributions from our manual calculations of gene set scores (mean of all genes in the gene set). 
+Again, there is a bimodal distribution for `RIGGI` and `ZHANG`, but not for `SILIGAN` and it is only present in `SCPCL000822`. 
+
+Let's use the local minima from the `RIGGI` gene set scores for `SCPCL000822` to identify tumor cells. 
+First we will look at how using `UCell` compares to our previous classifications for `SCPCL000822`. 
+
+```{r}
+# create distribution 
+density_data <- density(ref_ucell$RIGGI_EWING_SARCOMA_PROGENITOR_UP_UCell)
+# find the local minima in the distribution
+geneset_cutoff <- optimize(approxfun(density_data$x, density_data$y), interval = c(0.01, 0.2))$minimum
+
+# get a vector of tumor cells from ref 
+ref_ucell_tumor_cells <- ucell_df |> 
+  dplyr::filter(gene_list == "RIGGI",
+                signature_score >= geneset_cutoff,
+                sample == "SCPCL000822") |> 
+  dplyr::pull(barcodes) |> 
+  unique()
+
+# add classification column for comparison to original classification
+caret_df <- caret_df |> 
+  dplyr::mutate(
+    ucell_classification = dplyr::if_else(barcodes %in% ref_ucell_tumor_cells, "Tumor", "Normal")
+  )
+
+# confusion matrix between original and ucell 
+caret::confusionMatrix(
+  table(
+    caret_df$tumor_cell_classification, 
+    caret_df$ucell_classification 
+  )
+)
+```
+
+It looks like we get pretty good agreement between using `UCell` with the `RIGGI` gene set to the original classification. 
+Now we will use the cutoff from `SCPCL000822` to classify tumor cells in `SCPCL000824`. 
+
+```{r}
+# get list of tumor cells using ucell 
+new_ucell_tumor_cells <- ucell_df |> 
+  dplyr::filter(gene_list == "RIGGI",
+                signature_score >= geneset_cutoff,
+                sample == "SCPCL000824") |> 
+  dplyr::pull(barcodes) |> 
+  unique()
+
+# add ucell classification
+classification_df <- classification_df |> 
+  dplyr::mutate(
+    ucell_classification = dplyr::if_else(barcodes %in% new_ucell_tumor_cells, "Tumor", "Normal")
+  )
+
+ggplot(classification_df, aes(x = UMAP1, y = UMAP2, color = ucell_classification)) +
+  geom_point(alpha = 0.5, size = 0.5)
+```
+
+Here we see less cells classified as tumor cells than we saw with the `AUCell` classification. 
+This also looks very similar to the original UMAP when we classified cells using marker gene expression. 
```

Comment:
Whatever UCell is doing, it's score do still seem to be much more sample-dependent than the AUCell scores, making transfering thresholds less appealing. The AUCell scores seem to be more consistent between samples, but it is hard to see precisely without pulling out the raw scores and plotting them together. It would be nice to have similar plots for that section if you could, because the default AUCell plots are kind of busy. Which might be fine, but without an explanation of what all the fits and lines are, it is hard to know for sure!